### DISCUSSION TITLE
Can Autogluon do Multi-variable pattern analysis?

### DISCUSSION THREAD

**Discussion by buer19970329** [autogluon/autogluon#1921] [Q&A] [answered]
Thanks for your fantastic work! 

1) I am wondering if autogluon can do multivariable pattern analysis? (i.e., variable is a vector) I didn't find any related information in your document. 

2) If the result of Autogluon showed that KNN got the best prediction, how can I get the hyperparameters to get similar results using the KNN package?

I would be very grateful if anyone could answer my question. thanks in advance

**Comment by Innixma** [comment, marked as answer]
1. I'm unfamiliar with MVPA. If MVPA leverages supervised learning (binary, multiclass, or regression) then the answer is probably yes. If the problem can be broken down into input/output definition, then it would be easier to answer.
2. This kind of question has been asked a lot, so I'm actually adding my answer to our FAQ on our website going forward.

### How can I reproduce the results of model XYZ without using AutoGluon?

Exactly reproducing the results of AutoGluon models without using AutoGluon is extremely challenging.
(Ex: Trying to replicate the results of AutoGluon's LightGBM without depending on AutoGluon)
AutoGluon uses a variety of techniques not present in the model frameworks it builds upon such as data preprocessing,
edge case handling, custom early stopping, and custom architectures. These are all used to enhance the performance
and simplify usage beyond what exists in the original model framework (such as LightGBM, XGBoost, FastAI, etc.).
This is a core benefit of AutoML frameworks and is inherently complicated to replicate
(you would have to reimplement much of what AutoGluon does behind the scenes).
If you still wish to try to replicate results, you must dive into the source code for a given model
without guidance from the AutoGluon devs.
